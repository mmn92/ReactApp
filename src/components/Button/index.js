import React, { Component } from "react";

export default class Button extends Component {
  constructor(props) {
    super(props);
    this.state = { text: "" };

    this.handleClick = this.handleClick.bind(this);
  }

  handleClick() {
    this.setState({ text: "Clicked" });
  }

  render() {
    return (
      <button onClick={this.handleClick}>
        {this.state.text || this.props.text}
      </button>
    );
  }
}
