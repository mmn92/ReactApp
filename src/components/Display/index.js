import React from "react";

export const Display = props => (
  <div>
    <h1>Hi, {props.name}!</h1>
  </div>
);
