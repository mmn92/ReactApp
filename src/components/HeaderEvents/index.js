import React from 'react';

import { Header, Title } from './styles';

const HeaderEvents = () => (
    <Header>
        <Title>HeaderTitle</Title>
    </Header>
);

export default HeaderEvents;