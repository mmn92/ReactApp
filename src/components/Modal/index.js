import React from "react";

export const Modal = props => {
  let className = "modal__card";

  if (props.isOpen) className += " modal__card--open";

  return (
    <div className={className}>
      <div className="modal__data">{props.children}</div>
      <button onClick={props.handleClose}> X </button>
    </div>
  );
};
