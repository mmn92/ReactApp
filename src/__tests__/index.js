import React from "react";
import { create } from "react-test-renderer";
import axios from "axios";

import { Display } from "../components/Display";
import Button from "../components/Button";
import Users from "../components/Users";

describe("Display component", () => {
  test("it matches snapshot", () => {
    const component = create(<Display name="Matheus" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
});

describe("Button component", () => {
  test("it changes the text when clicked", () => {
    const component = create(<Button text="default" />);
    const rootInstance = component.root;

    const button = rootInstance.findByType("button");
    button.props.onClick();

    expect(button.props.children).toBe("Clicked");
  });
});

jest.mock("axios");

describe("Users component", () => {
  it("shows a list of users", async () => {
    const response = {
      data: [{ name: "Mr Man" }, { name: "Lord Dude" }]
    };
    axios.get.mockResolvedValue(response);

    const component = create(<Users />);
    const instance = component.getInstance();
    await instance.componentDidMount();

    const root = component.root;
    const listOfLi = root.findAll(element => element.type === "li");

    expect(listOfLi[0].props.children).toBe("Mr Man");
    expect(listOfLi[1].props.children).toBe("Lord Dude");
  });
});
