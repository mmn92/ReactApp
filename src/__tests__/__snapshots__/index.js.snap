// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display component it matches snapshot 1`] = `
<div>
  <h1>
    Hi, 
    Matheus
    !
  </h1>
</div>
`;
